File: GBDemo/PinManagement/DomainLayer/Pin/Pin.cs
using System;
using PinManagement.Helper.Domain;

namespace PinManagement.DomainLayer.Pin
{
    public class Pin: Entity<String>
    {
        public String PinId { get; private set; }
        public String Name { get; set; }
        public String SiteId { get; set; }


        public Pin(String PinId): base(PinId)
        {
            this.PinId = PinId;
         
        }
    }
}

File: guwave-onetest-v2/site/src/main/java/com/guwave/onetest/site/domain/pin/model/Pin.java
package com.guwave.onetest.site.domain.pin.model;

import com.guwave.onetest.common.domain.Entity;
import com.guwave.onetest.common.util.UUIDUtil;
import lombok.Getter;
import lombok.extern.log4j.Log4j2;
import org.springframework.lang.NonNull;

@Log4j2(topic = "Pin")
@Getter
public class Pin extends Entity<String> {
    private String name;

    public static Pin create(@NonNull String name) {
        return new Pin(UUIDUtil.uuid(), name);
    }

    public Pin(String id, String name) {
        super(id);
        this.name = name;
    }

    public void renamePin(String newName) {
        this.name = newName;
    }
}
